src/core/stream/handle_buffer_events.ts

```diff
@@ -0,0 +1,288 @@
+
+        default:
+          return defaultHandler(evt);
+    }
+  };
+
+  /**
+   * Call the right APIs when an "end-of-stream" event is received.
+   * @returns {Observable}
+   */
+  function handleEndOfStreamEvent() : Observable<
+    IEndOfStreamEvent |
+    IResumeStreamEvent
+  > {
+    const streamResumed$ = sharedPeriodBuffers$
+      .pipe(filter(({ type }) => type === "resume-stream"));
+
+    return maintainEndOfStream(mediaSource)
+      .pipe(ignoreElements(), takeUntil(streamResumed$));
+  }
+
+  /**
+   * @param {Object} needsStreamReloadEvt
+   * @returns {Observable}
+   */
+  function handleNeedsStreamReloadEvent(
+    needsStreamReloadEvt : INeedsStreamReloadEvent
+  ) : Observable<IReloadingStreamEvent|IStreamReloadedEvent> {
+    const { requestPause, clock$ } = args;
+    const { validateReload, bufferType } = needsStreamReloadEvt.value;
+
+    const isAlreadyAnnounced = !!currentBuffersReloading.length;
+    if (!arrayIncludes(currentBuffersReloading, bufferType)) {
+      currentBuffersReloading.push(bufferType);
+    }
+
+    return observableConcat(
+      // announce reload
+      isAlreadyAnnounced ? EMPTY : observableOf(EVENTS.reloadingStream()),
+
+      observableDefer(() => {
+        if (activeReloadPauseRequest == null) {
+          log.info("Stream: making pause request to reload content.");
+          activeReloadPauseRequest = requestPause();
+        }
+
+        const bufferReloaded$ = sharedPeriodBuffers$
+          .pipe(filter((evt) =>
+            evt.type === "added-segment" && evt.value.bufferType === bufferType ||
+            evt.type === "complete-buffer" && evt.value.type === bufferType
+          ));
+
+        const nudgeWhenReloaded$ = bufferReloaded$.pipe(
+          mergeMap((evt) => {
+            const indexOf = currentBuffersReloading.indexOf(bufferType);
+            if (indexOf < 0) {
+              log.error(`Stream: ${bufferType} buffer was not reloading`);
+              return EMPTY;
+            }
+
+            const originalTime = mediaElement.currentTime;
+            if (
+              evt.type !== "complete-buffer" &&
+              originalTime < 0.0001 ?
+                !isTimeInTimeRanges(mediaElement.buffered, originalTime + 0.0001) :
+                !isTimeInTimeRanges(mediaElement.buffered, originalTime - 0.0001)
+            ) {
+              return EMPTY;
+            }
+
+            currentBuffersReloading.splice(indexOf, 1);
+            if (currentBuffersReloading.length) {
+              return EMPTY;
+            }
+
+            // When buffer cleaned and reloaded, perform mini-seeks to flush the
+            // previous decoder's buffer
+            if (mediaElement.ended) {
+              return observableOf({ seeked : false });
+            }
+
+            mediaElement.currentTime = originalTime < 0.0001 ?
+              originalTime + 0.0001 : originalTime - 0.0001;
+            mediaElement.currentTime = originalTime;
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Verify this is the index that was intended; "currentTime" was already set on line 181. ([typescript:S4143](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4143))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=318&issues=AWYbSRWEM63is4RDZrpC&open=AWYbSRWEM63is4RDZrpC)

[ISSUE_KEY]: AWYbSRWEM63is4RDZrpC